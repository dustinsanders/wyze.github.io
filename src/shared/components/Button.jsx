// @flow

import type { Style, VNode } from '../../types'
import { css, hover } from 'glamor'
import H2 from './H2'
import preact from 'preact'

type Props = {
  text: string,
}

const color: Style = '#f4f6ff'
const button: Style = css(
  {
    backgroundColor: 'transparent',
    borderRadius: 5,
    border: `1px solid ${color}`,
    color,
    cursor: 'pointer',
    outline: 'none',
    marginTop: '2em',
    padding: '1em 3em',
    transitionDuration: '.5s',
  },
  hover({
    borderColor: '#8bc34a',
    color: '#8bc34a',
  }),
)

// Can safely link to `/` here because there is only one page.
const Button = ({ text }: Props): VNode => (
  <a href="/">
    <button {...button} type="button">
      <H2>{text}</H2>
    </button>
  </a>
)

export default Button
