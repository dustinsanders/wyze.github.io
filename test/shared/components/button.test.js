// @flow

import Button from '../../../src/shared/components/Button'
import snapshot from '../../helpers/snapshot'

describe('<Button />', () => {
  it('renders', snapshot.bind(null, Button))
  it('renders with text', snapshot.bind(null, Button, { text: 'Try again' }))
})
