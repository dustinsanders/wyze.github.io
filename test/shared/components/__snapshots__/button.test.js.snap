// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders 1`] = `
"<a href=\\"/\\">
  <button data-css-1fdb9qj type=\\"button\\">
    <H2></H2>
  </button>
</a>"
`;

exports[`<Button /> renders with text 1`] = `
"<a href=\\"/\\">
  <button data-css-1fdb9qj type=\\"button\\">
    <H2>Try again</H2>
  </button>
</a>"
`;
